library/data/registry-item-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/impl/RegistryItemAttributeReloader.java

```diff
@@ -0,0 +1,170 @@
+import org.quiltmc.qsl.resource.loader.api.reloader.SimpleResourceReloader;
+import net.minecraft.resource.Resource;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.InvalidIdentifierException;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+public final class RegistryItemAttributeReloader implements SimpleResourceReloader<RegistryItemAttributeReloader.LoadedData> {
+	private static final Logger LOGGER = LogManager.getLogger("AttributeReloader");
+	private static final Identifier ID = new Identifier("quilt", "attributes");
+
+	@Override
+	public Identifier getQuiltId() {
+		return ID;
+	}
+
+	@Override
+	public CompletableFuture<LoadedData> load(ResourceManager manager, Profiler profiler, Executor executor) {
+		return CompletableFuture.supplyAsync(() -> {
+			Map<RegistryItemAttribute<?, ?>, AttributeMap> attributeMaps = new HashMap<>();
+			for (Map.Entry<? extends RegistryKey<? extends Registry<?>>, ? extends Registry<?>> entry : Registry.REGISTRIES.getEntries()) {
+				Identifier registryId = entry.getKey().getValue();
+				StringBuilder pathSB = new StringBuilder();
+				if (!"minecraft".equals(registryId.getNamespace()))
+					pathSB.append(registryId.getNamespace()).append('/');
+				pathSB.append(registryId.getPath());
+				profiler.push(ID + "/finding_resources/" + pathSB);
+				Collection<Identifier> jsonIds = manager.findResources("attributes/" + pathSB, s -> s.endsWith(".json"));
+				if (jsonIds.isEmpty())
+					continue;
+				profiler.pop();
+				BuiltinRegistryItemAttributeHolder<?> holder = RegistryItemAttributeHolderImpl.getBuiltin(entry.getValue());
+				for (Identifier jsonId : jsonIds) {
+					Identifier attribId = getAttributeId(jsonId);
+					RegistryItemAttribute<?, ?> attrib = holder.getAttribute(attribId);
+					if (attrib == null) {
+						LOGGER.warn("Unknown attribute {} (from {})", attribId, jsonId);
+						continue;
+					}
+					profiler.push(ID + "/getting_resources{" + jsonId + "}");
+					List<Resource> resources;
+					try {
+						resources = manager.getAllResources(jsonId);
+					} catch (IOException e) {
+						LOGGER.error("Failed to get all resources for {}", jsonId);
+						LOGGER.catching(Level.ERROR, e);
+						continue;
+					}
+					AttributeMap attribMap = attributeMaps.computeIfAbsent(attrib,
+							id -> new AttributeMap(entry.getValue(), attrib));
+					profiler.swap(ID + "/processing_resources{" + jsonId + "," + attribId + "}");
+					for (Resource resource : resources) {
+						attribMap.processResource(resource);
+					}
+					profiler.pop();
+				}
+			}
+			return new LoadedData(attributeMaps);
```

Comment:
This long block of text makes the method unreadable. Could you add some empty lines to separate logical blocks of code?

Reply:
In the old codestyle there was mandatory blank lines between different indentation levels.

Reply:
I added comments. Does that work?

Reply:
That's a bit better, but imo you should still add some blank lines

Reply:
Nice! It's much more readable now.